client/extensions/woocommerce/app/dashboard/widgets/stats-widget/index.js

```diff
@@ -0,0 +1,363 @@
+		const date = getStartPeriod( moment().format( 'YYYY-MM-DD' ), unit );
+		const delta = getDeltaFromData( visitorData, date, 'visitors', unit );
+		return (
+			<Stat
+				site={ site }
+				label={ translate( 'Visitors' ) }
+				data={ visitorData || [] }
+				delta={ delta }
+				date={ date }
+				statType="statsVisits"
+				query={ queries.visitorQuery }
+				attribute="visitors"
+				type="number"
+			/>
+		);
+	};
+
+	renderConversionRate = () => {
+		const { site, translate, unit, visitorData, orderData, queries } = this.props;
+		const date = getUnitPeriod( moment().format( 'YYYY-MM-DD' ), unit );
+		const data = getConversionRateData( visitorData, orderData.data, unit );
+		const delta = getDeltaFromData( data, date, 'conversionRate', unit );
+		return (
+			<Stat
+				site={ site }
+				label={ translate( 'Conversion rate' ) }
+				data={ data || [] }
+				delta={ delta }
+				date={ date }
+				statType="statsOrders"
+				query={ queries.orderQuery }
+				attribute="conversionRate"
+				type="percent"
+			/>
+		);
+	};
+
+	renderReferrers = () => {
+		const { site, translate, unit, referrerData, queries } = this.props;
+		const { referrerQuery } = queries;
+
+		const row = find( referrerData, d => d.date === referrerQuery.date );
+		const fetchedData =
+			( row && sortBy( row.data, r => -r.sales ).slice( 0, dashboardListLimit ) ) || [];
+
+		const values = [
+			{ key: 'referrer', title: translate( 'Referrer' ), format: 'text' },
+			{ key: 'product_views', title: translate( 'Referrals' ), format: 'text' },
+			{ key: 'sales', title: translate( 'Sales' ), format: 'currency' },
+		];
+
+		const emptyMessage =
+			'day' === unit
+				? translate(
+						'Data is being processed. Switch to the week or month view to see your latest referrers.'
+					)
+				: translate( 'No referral activity has been recorded for this time period.' );
+
+		return (
+			<List
+				site={ site }
+				statSlug="referrers"
+				statType="statsStoreReferrers"
+				unit={ unit }
+				values={ values }
+				query={ referrerQuery }
+				fetchedData={ fetchedData }
+				emptyMessage={ emptyMessage }
+			/>
+		);
+	};
+
+	renderProducts = () => {
+		const { site, translate, unit, topEarnersData, queries } = this.props;
+		const { topEarnersQuery } = queries;
+		const values = [
+			{ key: 'name', title: translate( 'Product' ), format: 'text' },
+			{ key: 'total', title: translate( 'Sales' ), format: 'currency' },
+		];
+
+		return (
+			<List
+				site={ site }
+				statSlug="products"
+				statType="statsTopEarners"
+				unit={ unit }
+				values={ values }
+				query={ topEarnersQuery }
+				fetchedData={ topEarnersData }
+				emptyMessage={ translate( 'No products have been sold in this time period.' ) }
+			/>
+		);
+	};
+
+	queryData = () => {
+		const { site, queries } = this.props;
+		const { orderQuery, topEarnersQuery, referrerQuery, visitorQuery } = queries;
+		return (
+			<Fragment>
+				<QueryPreferences />
```

Comment:
I haven't taken this for a spin yet, but does `site` always exist? I'm not familiar with `getSelectedSiteWithFallback` just yet

Reply:
> I haven't taken this for a spin yet, but does site always exist? 

It should, because the app wrapper [doesn't render anything until site is available](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/app/index.js#L86). However, I can add a check here just to be safe.

Reply:
> Quite a few new queries - but fortunately most are against the wpcom api.

Yeah. All the stats ones are against .com and the synced data. The preferences one is also a .com endpoint. So no extra Jetpack/Proxy calls.